plugins/flytekit-wandb/README.md

```diff
@@ -0,0 +1,89 @@
+# Flytekit Weights & Biases Plugin
+
+The Weights & Biases MLOps platform helps AI developers streamline their ML workflow from end-to-end. This plugin
+enables seamless use of Weights & Biases within Flyte by configuring links between the two platforms.
+
+To install the plugin, run:
+
+```bash
+pip install flytekitplugins-wandb
+```
+
+Here is an example of running W&B with XGBoost using W&B for tracking:
+
+```python
+from flytekit import task, Secret, ImageSpec, workflow
+
+from flytekitplugins.wandb import wandb_init
+
+WANDB_PROJECT = "flytekit-wandb-plugin"
+WANDB_ENTITY = "github-username"
+WANDB_SECRET_KEY = "wandb-api-key"
+WANDB_SECRET_GROUP = "wandb-api-group"
+REGISTRY = "localhost:30000"
+
+image = ImageSpec(
+    name="wandb_example",
+    python_version="3.11",
+    packages=["flytekitplugins-wandb", "xgboost", "scikit-learn"],
+    registry=REGISTRY,
+)
+
+
+@task(
+    container_image=image,
+    secret_requests=[Secret(key=WANDB_SECRET_KEY, group=WANDB_SECRET_GROUP)],
+)
+@wandb_init(
+    project=WANDB_PROJECT,
+    entity=WANDB_ENTITY,
+    secret_key=WANDB_SECRET_KEY,
```

Comment:
Is it necessary to specify the secret key two times? (`@task` and `@wandb_init`)

Reply:
I agree, you can simply use the one specified in the task. as it will be part of the context

Reply:
I updated the PR to pass the secret object around:

```python
wandb_secret = Secret(key=WANDB_SECRET_KEY, group=WANDB_SECRET_GROUP)

@task(
    container_image=image,
    secret_requests=[wandb_secret],
)
@wandb_init(
    project=WANDB_PROJECT,
    entity=WANDB_ENTITY,
    secret=wandb_secret,
)
```

We can do something more implicit by pulling the secret request out of the task, but the `wandb_init` decorator still needs to know the name of the Secret associated with W&B. I prefer the more explicit approach of passing the `Secret` object around.